test/support/generators.ex

```diff
@@ -0,0 +1,293 @@
+
+      "tuple" ->
+        StreamData.list_of(scalar(), max_length: 5) |> StreamData.map(&List.to_tuple/1)
+
+      "list" ->
+        StreamData.list_of(scalar(), max_length: 5)
+
+      "map" ->
+        StreamData.map_of(
+          StreamData.one_of([StreamData.binary(), StreamData.integer()]),
+          scalar(),
+          max_length: 5
+        )
+    end)
+    |> StreamData.one_of()
+  end
+
+  defp data(depth, opts) do
+    using = Keyword.get(opts, :using, [])
+    excluding = Keyword.get(opts, :excluding, [])
+
+    filter =
+      case using do
+        [] -> {:reject, excluding}
+        _ -> {:accept, using}
+      end
+
+    ["integer", "boolean", "null", "string", "list", "tuple", "map"]
+    |> then(fn kinds ->
+      case filter do
+        {:reject, excluding} -> Enum.reject(kinds, &Enum.member?(excluding, &1))
+        {:accept, using} -> Enum.filter(kinds, &Enum.member?(using, &1))
+      end
+    end)
+    |> Enum.map(fn
+      "integer" ->
+        StreamData.integer()
+
+      "boolean" ->
+        StreamData.boolean()
+
+      "null" ->
+        StreamData.constant(nil)
+
+      "string" ->
+        StreamData.binary()
+
+      "tuple" ->
+        StreamData.list_of(data(depth + 1), max_length: 5) |> StreamData.map(&List.to_tuple/1)
+
+      "list" ->
+        StreamData.list_of(data(depth + 1), max_length: 5)
+
+      "map" ->
+        StreamData.map_of(
+          StreamData.one_of([StreamData.binary(), StreamData.integer()]),
+          data(depth + 1),
+          max_length: 5
+        )
+    end)
+    |> StreamData.one_of()
+  end
+
+  defp schematic_from_data(data) when is_integer(data),
+    do:
+      StreamData.member_of([
+        int(),
+        int(data),
+        oneof([int(), simple_schematic() |> Enum.fetch!(1)]),
+        oneof([int(data), simple_schematic() |> Enum.fetch!(1)])
+      ])
+
+  defp schematic_from_data(data) when is_binary(data),
+    do:
+      StreamData.member_of([
+        str(),
+        str(data),
+        oneof([str(), simple_schematic() |> Enum.fetch!(1)]),
+        oneof([str(data), simple_schematic() |> Enum.fetch!(1)])
+      ])
+
+  defp schematic_from_data(data) when is_boolean(data),
+    do:
+      StreamData.member_of([
+        bool(),
+        bool(data),
+        oneof([bool(), simple_schematic() |> Enum.fetch!(1)]),
+        oneof([bool(data), simple_schematic() |> Enum.fetch!(1)])
+      ])
+
+  defp schematic_from_data(data) when is_nil(data),
+    do: StreamData.member_of([null(), nullable(simple_schematic() |> Enum.fetch!(1))])
+
+  defp schematic_from_data(data) when is_tuple(data) do
+    Tuple.to_list(data)
+    |> Enum.map(fn datum -> schematic_from_data(datum) |> Enum.fetch!(1) end)
+    |> tuple()
+    |> StreamData.constant()
+  end
+
+  defp schematic_from_data(data) when is_list(data) do
+    one_of_schematic =
+      Enum.reduce(data, [], fn datum, acc ->
+        [schematic_from_data(datum) |> Enum.fetch!(1) | acc]
+      end)
+      |> oneof()
```

Comment:
because StreamData and schematic have a similar API, i wonder if we should avoid importing Schematic in this module and have everything be a remote call `Schematic.oneof()`

Reply:
i could go either way, but the latter is certainly more clear. i'll drop the import